Adds `execute` api method. (#39781)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/userscripts/execute/index.md

---
title: userScripts.execute()
slug: Mozilla/Add-ons/WebExtensions/API/userScripts/execute
page-type: webextension-api-function
browser-compat: webextensions.api.userScripts.execute
---

{{AddonSidebar}}

> Injects a user script defined in the `js` property into a target context (e.g., tab or frame).

Reviewer: ```suggestion
Injects a user script into a target context (e.g., tab or frame).
```
Should this information from the Chrome documentation be included? "By default, the script is run at `document_idle`, or immediately if the page is loaded. If the `injectImmediately` property is set, the script injects without waiting, even if the page has not finished loading.

## Syntax

```js
let executeUserScript = chrome.userScripts.execute(
  injection, // array of objects
);
```

### Parameters

- `injection`

>   - : `Array`of {{WebExtAPIRef("userScripts.InjectionUserScript")}}. Defines where and what user scripts to inject.

Reviewer: @Rob--W @dotproto are any of `InjectionUserScript`, `InjectionResult`. or `InjectionTarget` likely to be reused or can these be documented in line?

### Return Value

> A {{JSxRef("Promise")}} that consists of an array of {{WebExtAPIRef("userScripts.InjectionResult")}} objects containing the outcome of the injection.

Reviewer: Please add details of the outcome if the injection fails.
```suggestion
A {{JSxRef("Promise")}} fulfilled with an array of {{WebExtAPIRef("userScripts.InjectionResult")}} objects.
```

---
title: userScripts.execute()
slug: Mozilla/Add-ons/WebExtensions/API/userScripts/execute
page-type: webextension-api-function
browser-compat: webextensions.api.userScripts.execute
---

{{AddonSidebar}}

Injects a user script into a target context (e.g., tab or frame).

> [!NOTE]
> By default, the injected script is executed at `document_idle`, or immediately if the page has already loaded. If the `injectImmediately` property is set to `true`, the script injects without waiting, even if the page has not finished loading.

## Syntax

```js
let executeUserScript = browser.userScripts.execute(
  injection, // array of objects
);
```

### Parameters

- `injection`
  - : `Array` of {{WebExtAPIRef("userScripts.InjectionUserScript")}}. Defines where and what user scripts to inject.

### Return Value

> A {{JSxRef("Promise")}} fulfilled with an array of {{WebExtAPIRef("userScripts.InjectionResult")}} objects. If the injection fails, the attached result object will include an `error` property.

Reviewer: ```suggestion
A {{JSxRef("Promise")}} fulfilled with an array of {{WebExtAPIRef("userScripts.InjectionResult")}} objects. If the injection fails, the result object includes an `error` property.
```

page-type: webextension-api-function
browser-compat: webextensions.api.userScripts.execute
---

{{AddonSidebar}}

Injects a user script into a target context (e.g., tab or frame).

> [!NOTE]
> By default, the injected script is executed at `document_idle`, or immediately if the page is loaded. If the `injectImmediately` property is set to `true`, the script injects without waiting, even if the page is loading.

## Syntax

```js
let executeUserScript = browser.userScripts.execute(
  injection, // array of objects
);
```

### Parameters

- `injection`
  - : An array of objects specifying what user scripts to inject, where, and how.
    - `injectImmediately` {{optional_inline}}
      - : `boolean`. If set to `true`, the injection is triggered as soon as possible in the target context. This doesn't guarantee that the injection occurs before the page finishes loading, as the page may load before the script has reached the target.

    - `js`
      - : `array` of {{WebExtAPIRef("userScripts.ScriptSource")}}. The scripts to inject into matching pages.
    - `target`
      - : An object that defines the target context scripts are injected into.

> ` ``

Reviewer: ```suggestion
```
